src/sessionDappService.ts

```diff
@@ -297,18 +304,18 @@ export class DappService {
     })
   }
 
-  // TODO: keep for future developments
-
-  /* public async getOutsideExecutionCall(
+  public async getOutsideExecutionCall(
     sessionRequest: OffChainSession,
     sessionAuthorizationSignature: ArraySignatureType,
+    cacheAuthorisation: boolean,
     calls: Call[],
     revision: TypedDataRevision,
     accountAddress: string,
+    provider: ProviderInterface | Provider,
     caller = "ANY_CALLER",
     execute_after = 1,
     execute_before = 999999999999999,
-    nonce = randomStarknetKeyPair().publicKey,
+    nonce: BigNumberish,
```

Comment:
i think using random numbers as a default was great, because that's what they want 99% of the time